src/app.ts

```diff
@@ -0,0 +1,72 @@
+import { INestApplication, ValidationPipe, VersioningType } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { TransformInterceptor } from '@ukef/helpers';
+import { SwaggerDocs } from '@ukef/swagger';
+import compression from 'compression';
+import { Logger, LoggerErrorInterceptor } from 'nestjs-pino';
+
+import { AcbsExceptionTransformInterceptor } from './modules/acbs-adapter/acbs-exception-transform.interceptor';
+
+export class App {
+  private readonly configService: ConfigService;
+  public readonly port: number;
+
+  constructor(protected readonly app: INestApplication) {
+    this.configService = app.get<ConfigService>(ConfigService);
+    this.port = this.getConfig<number>('PORT') || 3003;
+
+    const env: string = this.getConfig<string>('app.env');
+    process.env.NODE_ENV = env;
+
+    const globalPrefix: string = this.getConfig<string>('app.globalPrefix');
+    const version: string = this.getConfig<string>('app.versioning.version');
+    const versioningPrefix: string = this.getConfig<string>('app.versioning.prefix');
+    app.enableVersioning({
+      type: VersioningType.URI,
+      defaultVersion: version,
+      prefix: versioningPrefix,
+    });
+
+    app.setGlobalPrefix(globalPrefix);
+    app.useGlobalPipes(
+      new ValidationPipe({
+        whitelist: true,
+        transform: true,
+        forbidNonWhitelisted: true,
+        transformOptions: {
+          enableImplicitConversion: true,
+        },
+      }),
+    );
+
+    // Swagger docs
+    SwaggerDocs(app);
+
+    app.useGlobalInterceptors(new TransformInterceptor());
+    app.useGlobalInterceptors(new LoggerErrorInterceptor());
+    app.useGlobalInterceptors(new AcbsExceptionTransformInterceptor());
```

Comment:
This file is extracted from `main.ts` - the only addition is this new global interceptor (`AcbsExceptionTransformInterceptor`)

Reply:
right. can I check why we need this file? if it's pretty much the same as `main.ts`?

Reply:
Yes, I kept it almost exactly the same as `main.ts` was before. The reason I extracted this from `main` was so that we could re-use this code when we create the app in the API tests

Reply:
ok cool. that's fine. the file names may get confusing as both `main` and `app` can indicate they are the entry points. 
we can leave it as it is for now